garak/generators/huggingface.py

```diff
@@ -492,14 +516,27 @@ def _call_model(
         if self.top_k is not None:
             self.generation_config.top_k = self.top_k
 
-        text_output = []
+        raw_text_output = []
         with warnings.catch_warnings():
             warnings.simplefilter("ignore", category=UserWarning)
             with torch.no_grad():
+                if self.use_chat:
+                    formatted_prompt = self.tokenizer.apply_chat_template(
+                        self._format_chat_prompt(prompt),
+                        tokenize=False,
+                        add_generation_prompt=True,
+                    )
+                else:
+                    formatted_prompt = prompt
+
                 inputs = self.tokenizer(
-                    prompt, truncation=True, return_tensors="pt"
+                    formatted_prompt, truncation=True, return_tensors="pt"
                 ).to(self.device)
```

Comment:
Note while this call to the tokenizer may be possible to optimize out a direct call to `self.tokenizer.apply_chat_template()` with `return_tensors` passed returns a different format that cannot be directly passed to `self.model.generate`, by taking this extra step the code is more consistent with and without chat support on the model.

Reply:
grumble grumble but yes I support you

Reply:
i like exerting this level of control over when tokenisation happens